examples/tree_2d_dgsem/elixir_diffusion_linear_map.jl

```diff
@@ -0,0 +1,111 @@
+using Trixi
+
+###############################################################################
+
+# Build pure diffusion (Laplace) operator
+advection_velocity = (0, 0)
+equations = LinearScalarAdvectionEquation2D(advection_velocity)
+diffusivity() = 1
+equations_parabolic = LaplaceDiffusion2D(diffusivity(), equations)
+
+# Hyperbolic flux does not matter for this example
+solver = DGSEM(polydeg = 5, surface_flux = flux_central)
+
+coordinates_min = (0.0, 0.0)
+coordinates_max = (1.0, 1.0)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 2,
+                n_cells_max = 80_000,
+                periodicity = false)
+
+# Define arbitrary initial condition,
+# required only for construction of the semidiscretization, but never called
+initial_condition = Returns(0)
+
+function bc_homogeneous(x, t, equations)
+    return SVector(0)
+end
+bc_homogeneous_dirichlet = BoundaryConditionDirichlet(bc_homogeneous)
+
+function bc_sin(x, t, equations)
+    return SVector(sinpi(2 * x[1]))
+end
+bc_sin_dirichlet = BoundaryConditionDirichlet(bc_sin)
+
+# Same boundary conditions for hyperbolic and parabolic part
+boundary_conditions = (; x_neg = bc_homogeneous_dirichlet,
+                       y_neg = bc_sin_dirichlet,
+                       y_pos = bc_sin_dirichlet,
+                       x_pos = bc_homogeneous_dirichlet)
+
+semi = SemidiscretizationHyperbolicParabolic(mesh,
+                                             (equations, equations_parabolic),
+                                             initial_condition, solver;
+                                             solver_parabolic = ViscousFormulationLocalDG(),
+                                             boundary_conditions = (boundary_conditions,
+                                                                    boundary_conditions))
+
+A_map, b = linear_structure(semi)
+
+# Direct solve, with explicit matrix construction.
+# Has some troubles due to poor conditioning, visible in top right corner
+#=
+A_matrix = Matrix(A_map) # This is very memory consuming
+u_ls = A_matrix \ b
+=#
+
+# Iterative solve, works directly on the linear map, no explicit matrix construction required!
+using Krylov
+
+u_ls, stats = gmres(A_map, b)
+
+###############################################################################
+
+# Construct the ODE problem for easy plotting and comparison to analytical solution.
+tspan = (0.0, 0.0)
+
+# Analytical/continuous solution
+function continuous_solution(x, t, equations)
+    a = (1 - cosh(2 * pi)) / (sinh(2 * pi))
+
+    u_sol = (cosh(2 * pi * x[2]) + a * sinh(2 * pi * x[2])) * sinpi(2 * x[1])
+    return SVector(u_sol)
+end
+initial_condition_sol = continuous_solution
+
+# Exchange only the initial condition, rest remains the same
```

Comment:
Maybe, I am missing something, but do we really _need_ two different semis? If we only construct the `semi_sol` based on the `continuous_solution` and pass that to `linear_structure`, the linear map wouldn't change, right? So what do we need `semi` with the zero initial condition for? Or do you deliberately want to clarify that the linear map "does not know" of the analytical solution for pedagogical reasons (which would be fine with me)?

Reply:
You are right, I actually learned from CodeCov that the `initial_condition` is only called in `semidiscretize`. So yes, this can be truncated to 1 semi :+1: